### DISCUSSION TITLE
Project Discussion

### DISCUSSION THREAD

**Discussion by flywire** [pfalcon/ppxml2db#14] [Debugging issues]
Can we have a project discussion channel?

For example, identifying the problem.

> If you are interested in a missing feature to be supported, or a conversion issues to be addressed, please prepare a standalone, small XML file demonstrating a problem and submit a ticket to the project bugtracker: https://github.com/pfalcon/ppxml2db/issues/ .

The user needs to understand what the problem is before they can prepare a demonstration file . btw, the repo doesn't contain any tests or even a _small XML file_.

I'm getting an error as follows but I don't understand why so I'd appreciate some guidance on how to identify the problem so I can't prepare a small demonstration:

```
DEBUG:dbhelper:INSERT OR REPLACE INTO xact(uuid, date, amount, shares, updatedAt, type, currency, _xmlid, account, acctype, _order, security, fees, taxes) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ['815a5c54-0f7f-4226-a5d8-e6f0da543a0f', '2023-01-05T00:00', '42416', '336000000000', '2023-01-12T20:44:16.061248100Z', 'DIVIDENDS', 'AUD', '149', '801d1bef-bc2d-4ec7-a7ce-390e6b0f2406', 'account', 426614, 'b9da1e5b-362d-4f16-9017-5f2fc1866597', 0, 0]
Traceback (most recent call last):
  File "E:\sqlite\ppxml2db.py", line 579, in <module>
    conv.iterparse()
  File "E:\sqlite\ppxml2db.py", line 481, in iterparse
    assert self.uuid2ctr_map[self.cur_uuid()].startswith("account")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
```

I understand the file is processed sequentially so the error must occur after the debug line. Parsing fails in a `crossEntry` object (#8) and it's a private file.

`print(self.uuid2ctr_map[self.cur_uuid()])` inserted before the asset error returns: `portfolio`

The main methodology I use is to open file in PP and progressively delete objects then run the file until it parses successfully.

* despite [54ad8f4](https://github.com/pfalcon/ppxml2db/commit/54ad8f43c342078b0f97191e8169e74299fda590) it seems portfolio transfers aren't supported yet.

**Comment by pfalcon** [comment]
> Can we have a project discussion channel?

Enabled "Discussions" in project, moving this topic there.

**Comment by pfalcon** [comment]
> btw, the repo doesn't contain any tests or even a small XML file.

Internally, I use adhoc testsuite based on samples you previously provided. I'd like to make more reusable, the issues:

1. Wanted to ask you if you'd be ok to make the previous samples fully public, i.e. they don't contain any traces of private/personal info. (They're kinda already public, as you posted them where they're publicly accessible, but it's one thing to have something dug in comments on a random issue, and another to ship those files with repo to anyone).
2. Many samples are fairly big comparing to the code itself, so I'm not sure I'm comfortable with such big drops.

**Reply by flywire** [reply]
> adhoc testsuite

I was wondering if the classes need unit tests.

> 1. ok to make the previous samples fully public?

Yes

> 2. Many samples are fairly big

Deleting most if not all price history likely significantly reduces file size.

**Comment by pfalcon** [comment]
> despite [54ad8f4](https://github.com/pfalcon/ppxml2db/commit/54ad8f43c342078b0f97191e8169e74299fda590) it seems portfolio transfers aren't supported yet.

Testcases which you provided in https://github.com/pfalcon/ppxml2db/issues/8 work. Well, testcases which you provided in #8 and which I have locally. I closed and locked it when I noticed that it takes more time/effort from me to constantly rescan long thread there to see what was done and what not, than actually working on fixing issues.

So, if there's XML which doesn't work, open a separate ticket with it (now should be in XML-id format of course).

**Comment by pfalcon** [comment]
>     assert self.uuid2ctr_map[self.cur_uuid()].startswith("account")
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AssertionError

Well, that means that the context for particular element is not what the app expects. As usual, you can always remove "assert" kw, but the tool likely fails elsewhere or produces invalid file.

> The main methodology I use is to open file in PP and progressively delete objects then run the file until it parses successfully.

You'd need to add print()s to identify which particular transaction causes the error, then delete everything else (most of everything else).

**Comment by pfalcon** [comment]
> > adhoc testsuite  
> 
> I was wondering if the classes need unit tests.

No.
